### PR TITLE
Add otel4s to registry

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

**Comment by rossabaker** [comment]
Drafted so @iRevive and @zmccoy can confirm spelling variants and whether they want their e-mails listed.  Sorted alphabetically, though Maksym deserves to be at the top. :laughing:

**Comment by iRevive** [comment]
Thanks! You can add my email: maxochenashko@gmail.com

File: data/registry/otel-scala.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+title: otel4s
+registryType: instrumentation
+isThirdParty: true # this is only true if the project is not maintained by the OpenTelemetry project
```
Comment (chalin) [inline data/registry/otel-scala.yml:3]: Drop the comment.

```suggestion
isThirdParty: true
```

**Review by chalin — COMMENTED** [review]
I know that this is still in draft, but in case I can't get back to this before it's approved and merged: ... drop the YAML comment

**Review by cartermp — APPROVED** [review]
Thanks!

**Pull request merged**